content/geodesic/module/usage/with-kops.md

```diff
@@ -1,117 +1,412 @@
+This assumes you've followed the [Geodesic Module Usage with Terraform]({{< relref "geodesic/module/usage/with-terraform.md" >}}) guide which covers all the scaffolding necessary to get started.
 {{% /dialog %}}
 
+Geodesic uses [kops]({{< relref "tools/kops.md" >}}) to manage kubernetes clusters.
+
 # Create a cluster
 
-Follow the [Provision a Cluster]({{< relref "geodesic/module/usage/with-kops.md" >}}) process
+Provisioning a `kops` cluster takes three steps:
 
-# Provision Platform Backing Services
+1. Provision a [terraform-aws-kops-state-backend]({{< relref "terraform-modules/kops-kubernetes/terraform-aws-kops-state-backend.md" >}}) which consists of an S3 bucket, cluster DNS zone, and SSH keypair to access the k8s masters and nodes.
+2. Update the `Dockerfile` and rebuild the Geodesic Module to generate a kops manifest file (and restart shell)
+3. Launch a kops cluster from the manifest file
 
-A number of [Terraform Modules Overview]({{< relref "terraform-modules/overview.md" >}}) provide to provision AWS resources needed by Charts like [external-dns](/kubernetes-backing-services/external-dns/) and [chart-repo]({{<relref "helm-charts/supported-charts/chart-repo.md" >}}). See our [Terraform modules for Kubernetes (Kops)](/terraform-modules/kops-kubernetes).
+## Provision the State Backend
 
-# Provisioning a Kops cluster
+### Config environment variables
+Add to the module `Dockerfile` environment variables
 
-We create a `kops` cluster from a manifest.
+```
+ENV KOPS_CLUSTER_NAME={KOPS_CLUSTER_NAME}
 
-The manifest template is located in [`/templates/kops/default.yaml`](https://github.com/cloudposse/geodesic/blob/master/rootfs/templates/kops/default.yaml)
-and is compiled by running the `build-kops-manifest` script as a `RUN` step in the `Dockerfile`.
+ENV TF_VAR_kops_cluster_name=${KOPS_CLUSTER_NAME}
+ENV TF_VAR_parent_zone_name={KOPS_CLUSTER_PARENT_DNS_ZONE_NAME}
+```
 
-# Synopsis
+Replace placeholders `{%}` with values specific for your project.
 
-Provisioning a `kops` cluster takes three steps:
+{{< dialog type="code-block" icon="fa fa-code" title="Example" >}}
+```
+ENV KOPS_CLUSTER_NAME=us-west-2.staging.example.com
 
-1. Provision a [terraform-aws-kops-state-backend]({{< relref "terraform-modules/kops-kubernetes/terraform-aws-kops-state-backend.md" >}}) which consists of an S3 bucket, cluster DNS zone, and SSH keypair to access the k8s masters and nodes.
-2. Update the `Dockerfile` and rebuild/restart the `geodesic` shell to generate a kops manifest file
-3. Launch a kops cluster from the manifest file
+ENV TF_VAR_kops_cluster_name=${KOPS_CLUSTER_NAME}
+ENV TF_VAR_parent_zone_name=staging.example.com
+```
+{{< /dialog >}}
+
+### Rebuild the module
+[Rebuild](/geodesic/module/usage/) the module
+```shell
+> make build
+```
+
+### Add kops state terraform module
+Create file in `./conf/aws-kops-backend/main.tf` with following content
+
+{{% dialog type="code-block" icon="fa fa-code" title="./conf/aws-kops-backend/main.tf" %}}
```

Comment:
I'd prefer that we move this code block to an external include prior to merging.

Reply:
https://github.com/cloudposse/docs/pull/64